spec/requests/api/allocations_controller_create_spec.rb

```diff
@@ -74,86 +77,48 @@
     let(:headers) { { 'CONTENT_TYPE': content_type }.merge('Authorization' => "Bearer #{access_token}") }
     let(:content_type) { ApiController::CONTENT_TYPE }
 
-    before do
-      next if RSpec.current_example.metadata[:skip_before]
-
-      post '/api/v1/allocations', params: { data: data }, headers: headers, as: :json
-    end
-
     context 'when not authorized', :with_invalid_auth_headers do
       let(:headers) { { 'CONTENT_TYPE': content_type }.merge(auth_headers) }
       let(:content_type) { ApiController::CONTENT_TYPE }
       let(:detail_401) { 'Token expired or invalid' }
 
+      before { post_allocations }
+
       it_behaves_like 'an endpoint that responds with error 401'
     end
 
     context 'with an invalid CONTENT_TYPE header' do
       let(:content_type) { 'application/xml' }
 
+      before { post_allocations }
+
       it_behaves_like 'an endpoint that responds with error 415'
     end
 
     context 'when successful' do
-      let(:allocation) { Allocation.find_by(from_location_id: from_location.id) }
+      before { post_allocations }
 
       it_behaves_like 'an endpoint that responds with success 201'
+    end
 
-      it 'creates a allocation', skip_before: true do
-        expect { post '/api/v1/allocations', params: { data: data }, headers: headers, as: :json }
-          .to change(Allocation, :count).by(1)
-      end
+    describe 'creating allocations' do
+      let(:allocation) { Allocation.find_by(from_location_id: from_location.id) }
 
-      it 'creates multiple moves', skip_before: true do
-        expect { post '/api/v1/allocations', params: { data: data }, headers: headers, as: :json }
-          .to change(Move, :count).by(2)
+      it 'creates an allocation' do
+        expect { post_allocations }.to change(Allocation, :count).by(1)
       end
 
       it 'audits the supplier' do
+        post_allocations
         expect(allocation.versions.map(&:whodunnit)).to eq([supplier.id])
       end
 
       it 'returns the correct data' do
+        post_allocations
         expect(response_json).to include_json resource_to_json
       end
 
-      it 'sets the correct number of complex_cases' do
-        expect(response_json.dig('data', 'attributes', 'complex_cases').size).to match complex_cases_attributes.size
-      end
-
-      it 'sets the correct complex_cases attributes' do
-        expect(response_json.dig('data', 'attributes', 'complex_cases').first).to match complex_case1_attributes.stringify_keys
-      end
-
-      it 'sets the correct status' do
```

Comment:
should this test be moved to the allocations creator spec?

Reply:
That's covered off by this spec:

```ruby
    it 'sets the correct attributes to an allocation' do
      call_creator
      expect(creator.allocation).to have_attributes(
        status: 'unfilled',
...
```